src/types/shared.ts

```diff
@@ -35,8 +35,14 @@ export interface IApplicationLogic {
 
 export interface IOnEntry {
   show: 'new-instance' | 'startpage' | string;
+  instanceSelection?: IInstanceSelection;
 }
 
+export type IInstanceSelection = {
+  rowsPerPageOptions: number[];
+  sortDirection: 'asc' | 'desc';
+};
+
```

Comment:
Boy scout rule! Prefer to clean up the duplication instead of maintaining these doubly.